examples/CMakeLists.txt

```diff
@@ -1,25 +1,53 @@
 option(GINKGO_BUILD_EXTLIB_EXAMPLE "Build the external-lib-interfacing with deal.II, you need to link the deal.II library." OFF)
-add_subdirectory(custom-logger)
-add_subdirectory(custom-matrix-format)
-add_subdirectory(custom-stopping-criterion)
+option(GINKGO_RUN_EXAMPLES " Compile run and validation targets for the examples." ON)
+
+set(EXAMPLES_EXEC_LIST
+    adaptiveprecision-blockjacobi
+    custom-logger
+    ginkgo-overhead
+    ginkgo-ranges
+    ilu-preconditioned-solver
+    ir-ilu-preconditioned-solver
+    inverse-iteration
+    iterative-refinement
+    mixed-precision-ir
+    nine-pt-stencil-solver
+    poisson-solver
+    preconditioned-solver
+    simple-solver
+    three-pt-stencil-solver)
+
+set(EXAMPLES_LIST
+    ${EXAMPLES_EXEC_LIST}
+    custom-matrix-format
+    custom-stopping-criterion
+    minimal-cuda-solver
+    papi-logging
+    performance-debugging
+    preconditioner-export
+    simple-solver-logging)
+
 if(GINKGO_BUILD_EXTLIB_EXAMPLE)
-    add_subdirectory(external-lib-interfacing)
+    list(APPEND EXAMPLES_LIST external-lib-interfacing)
+endif()
+
+if(GINKGO_RUN_EXAMPLES)
+    foreach(example ${EXAMPLES_LIST})
+        add_subdirectory(${example})
```

Comment:
I believe we should have this whether `GINKGO_RUN_EXAMPLES` is activated or not.